cs3api4lab/api/cs3_share_api.py

```diff
@@ -83,6 +79,22 @@ def _list(self):
             self._handle_error(list_response)
         return list_response
 
+    def list_grantees_for_file(self, file_path):
+        """
+        :param file_path: path to the file
+        :return: list of grantees
+        """
+        share_list = self.list()
+
+        file_path = ShareUtils.purify_file_path(file_path, self.config['client_id'])
+        shares = []
+        for share in share_list.shares:
+            path = ShareUtils.purify_file_path(share.resource_id.opaque_id, self.config['client_id'])
+            if file_path == path:
```

Comment:
So, you're getting all shares and then matching their paths to know which ones belong to a specific path, right?
Maybe, since we have all shares, I would also return the shares of the parent folders. This way we can provide this info to the user (ofc this needs some changes in the frontend so that we prevent users from updating those shares and to give a button to move to a parent.)
I'll open a new issue for that (we need a concept as well).

Reply:
this is basically a copy of a similar function in from share_api_facade, but sure, we can think about expanding this